### PR TITLE
basic support for batch requests

### PR DESCRIPTION
**Pull request by hugoArregui**
Add very basic support for batching calls in the same request: https://www.jsonrpc.org/specification#batch

I added a new `.toEthCall` to the SolidityFunction in order to build the payload for the call and the `sendBatchAsync` to the request manager in order to handle the batched requests. To be able to parse the results, I also exposed the `unpackOutput` in the SolidityFunction, which was private.

### CODE REVIEW

**Comment by menduz** [comment]
can you provide details in the description?

**Comment by hugoArregui** [comment]
> can you provide details in the description?

@menduz done, let me know if something is not clear please.

**Comment by menduz** [comment]
I haven't seen this on the wild actually, I'm afraid of the compatibility issues across different providers or nodes. The standard nowadays is to perform a [multicall](https://github.com/makerdao/multicall)

TLDR:

It is a smart contract call that accept a n-ary parameter of calls, then all of them are executed in the same RPC call and it returns an n-ary result (map)

**Comment by menduz** [comment]
@fzavalia @nachomazzara do you have a suggestion on how to proceed?

**Comment by nachomazzara** [comment]
I'm in favor of using an existing library. We are using [this](https://github.com/0xsequence/sequence.js/tree/master/packages/multicall) implementation in [decentraland-dapps](https://github.com/decentraland/decentraland-dapps/blob/master/src/modules/authorization/sagas.ts#L2)

**Comment by hugoArregui** [comment]
@nachomazzara are you using eth-connect as well? If you have an example of multicall with ethc-connect it will be great (don't worry if you don't)

**Comment by hugoArregui** [comment]
That been said, I think @menduz is talking about another option, supporting many operations at once in the contract, the library you shared @nachomazzara is an implementation of what I have done here.

**Comment by nachomazzara** [comment]
No, we don't. But, as @menduz said, it is a normal call you make to a smart contract. That smart contract can process multiple calls. So I think that should be as easy. 

If not, I recommend using ethers which is a library with great community support and the one that we are using in our dApps and servers.

**Comment by hugoArregui** [comment]
@nachomazzara thanks! I'll look into moving to ethers, I'm closing this PR, thanks!
